Fixes #1 Uses click to accept path for proposed endpoints doc

mansiag opened the pull request:
Uses click to Allow `serialize_data.py` to accept path for proposed endpoints docs as command line argument.

mansiag committed:
feat: allow serialize_data.py accept path for proposed endpoints doc as command line argument

CuriousLearner commented on src/serialize_data.py:
```diff
@@ -93,9 +94,11 @@ def get_json_from_endpoints(lines):
-def generate_json_from_docs_file(filename='proposed_endpoints.md'):
+@click.command()
+@click.argument('file_path', type=click.Path(exists=True))
+def generate_json_from_docs_file(file_path):
```
I believe that this path shouldn't be mandatory to provide. What do you think?

mansiag replied:
Yes, its better to have some default value of the path. I would make the necessary changes.

mansiag committed:
Fix: changes path argument from mandatory to optional

mansiag committed:
Add flake8

mansiag committed:
Create CHANGELOG.md

mansiag commented:
Can you suggest how should I document the feature in Readme

CuriousLearner commented:
State how do we execute the command with `--file_path` option.

mansiag committed:
Add: feature documentation

mansiag commented:
@CuriousLearner please review.

CuriousLearner commented on README.md:
```diff
@@ -32,7 +32,7 @@ pipenv install
 ## Usage
 
-Run `pipenv run python serialize_data.py` to generate JSON file named `endpoints_data.json`
+Run `pipenv run python serialize_data.py` to generate JSON file named `endpoints_data.json`. One can specify the path of the proposed endpoints docs  using `--file_path` option in the above command.
```
Ideally command line option should use `-` hyphens instead of underscore `_`

mansiag replied:
But it will give syntax error on using file-path as an argument to the generate_json_from_docs_file function.

CuriousLearner replied:
@mansiag I think there should be a way, to use `-` as part of the actual argument that receives the callback but also have the actual value come in as a valid identifier (that uses `_` to be consistent with the Python naming conventions).

Can you try finding a way?

mansiag replied:
Yeah Done.

CuriousLearner commented on CHANGELOG.md:
```diff
@@ -0,0 +1,9 @@
+Features & Improvements
+-----------------------
+
+- Allow `serialize_data.py` accept path for proposed endpoints docs as command line argument. [#1](https://github.com/CuriousLearner/API-stub-generator/pull/4)
```
Instead of URL to the issue, you can add your name and link to your GitHub profile.

CuriousLearner commented:
@mansiag 
Just minor nit-picks, rest looks good to me :)

mansiag committed:
Update CHANGELOG.md

Add Name and link to the GitHub profile of the contributor and remove link to the pull request.

mansiag committed:
follow command line options naming conventions

mansiag committed:
Merge branch 'master' of github.com:mansiag/API-stub-generator

CuriousLearner committed:
fix(docs): Update documentation

CuriousLearner reviewed (approved):
Hey, @mansiag Thanks for your work. I've handled minor documentation thing and it seems good to go!

Thanks for your patience and all your work :)
